main.go

```diff
@@ -59,6 +64,20 @@ func main() {
 		prometheus.MustRegister(certExporter)
 	}
 
+	// Monitor expiry of secrets of type kubernetes.io/tls
+	if monitorSecrets {
+		c := secret.DefaultConfig()
+		if namespaceWhitelist != "" {
+			c.Namespaces = strings.Split(namespaceWhitelist, ",")
+		}
+
+		secretExporter, err := secret.New(c)
+		if err != nil {
+			panic(microerror.Mask(err))
```

Comment:
Can we avoid panicing here?

Reply:
again - this is lifted from the existing code - i guess panicking is a bad idea as it stops the whole exporter working. i'll reduce the severity